### DISCUSSION TITLE
invite_link in chat_member listener

### DISCUSSION THREAD

**Discussion by cryptomadvad** [telegraf/telegraf#1697] [General]
Hi I am working on developing telegram bot. When new use joined the the supergroup, I want to know using which link user joined the group. 

In my test group, I am getting the invite_link object by using
`bot.on('chat_member', (ctx) > console.log(ctx.update.chat_member.invite_link)`

but when I try with 11k members group, I am not getting `invite_link` object.

Please let me know how can I get the invite_link with which user joined the group.

**Comment by silencebeat** [comment]
> Hi I am working on developing telegram bot. When new use joined the the supergroup, I want to know using which link user joined the group.
> 
> In my test group, I am getting the invite_link object by using `bot.on('chat_member', (ctx) > console.log(ctx.update.chat_member.invite_link)`
> 
> but when I try with 11k members group, I am not getting `invite_link` object.
> 
> Please let me know how can I get the invite_link with which user joined the group.

have you found the issue yet?

**Reply by MKRhere** [reply]
I believe you don't get an invite_link if the group is public. Can you test this hypothesis? Try in a private and public group.

**Reply by silencebeat** [reply]
> I believe you don't get an invite_link if the group is public. Can you test this hypothesis? Try in a private and public group.

yes, it doesn't return invite_link in public group. only private group
is there any way to get them in public group?

**Reply by hypernova7** [reply]
invite link only works on private channels/groups
